sklearn/preprocessing/data.py

```diff
@@ -2218,6 +2222,17 @@ def fit(self, X, y=None):
                                                        self.subsample))
 
         X = self._check_inputs(X)
+        n_samples = X.shape[0]
+
+        if self.n_quantiles > n_samples:
+            self.n_quantiles = n_samples
```

Comment:
It should be called `n_quantiles_` isn't it?

Reply:
yes indeed I just fixed it